platforms/android/library-compose/src/main/java/io/element/android/wysiwyg/compose/RichTextEditor.kt

```diff
@@ -40,15 +43,24 @@ fun RichTextEditor(
     state: RichTextEditorState = rememberRichTextEditorState(),
     registerStateUpdates: Boolean = true,
     style: RichTextEditorStyle = RichTextEditorDefaults.style(),
-    mentionDisplayHandler: MentionDisplayHandler? = null,
+    resolveMentionDisplay: ((text: String, url: String) -> TextDisplay)? = null,
+    resolveRoomMentionDisplay: (() -> TextDisplay)? = null,
```

Comment:
nit: The interface might be a bit clearer and more self documenting with a default implementation e.g.

```kotlin
(text: String, url: String) -> TextDisplay = RichTextEditorDefaults.MentionDisplay
```